domain/modelagent/service/service_test.go

```diff
@@ -618,3 +619,38 @@ func (s *suite) TestGetUnitReportedAgentVersionMissingAgentBinaries(c *gc.C) {
 	_, err := svc.GetUnitsAgentBinaryMetadata(context.Background())
 	c.Check(err, jc.ErrorIs, modelagenterrors.MissingAgentBinaries)
 }
+
+// TestSetAgentStreamNotValidAgentStream is testing that if we supply an
+// unknown agent stream to [Service.SetModelAgentStream] we get back an error
+// satisfying [coreerrors.NotValid].
+func (s *suite) TestSetAgentStreamNotValidAgentStream(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	// This is a fake stream that doesn't exist.
+	agentStream := coreagentbinary.AgentStream("bad value")
+
+	err := NewService(s.state).SetModelAgentStream(
+		context.Background(),
+		agentStream,
+	)
+	c.Check(err, jc.ErrorIs, coreerrors.NotValid)
+}
+
+// TestSetAgentStream is testing the happy path of setting the model's agent
+// stream.
+func (s *suite) TestSetAgentStream(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	agentStream := coreagentbinary.AgentStreamTesting
```

Comment:
Do we need this? Just  use the const directly

Reply:
Done & Done